app/src/ui/lib/section-filter-list.tsx

```diff
@@ -664,7 +684,19 @@ function createStateUpdate<T extends IFilterListItem>(
     selectedRow = getFirstVisibleRow(rows)
   }
 
-  return { rows: rows, selectedRow, filterValue: filter, groups: groupIndices }
+  let filterChanged = state.filterValueChanged
+
+  if (!filterChanged && filter.length) {
+    filterChanged = true
+  }
```

Comment:
We have an[ "Immutability is good" convention](https://github.com/desktop/desktop/blob/development/docs/contributing/engineering-values.md#immutability-is-good) we like to follow. It is not super hard and fast, but we strive for it when we can. Another undocumented convention is avoiding falsey checks and again not strongly adhered too. 

Think we could update this (in both list components) to something like :

```
const filterValueChanged = !state.filterValueChanged && filter.length > 0;
...
{
...
filterValueChanged,
}
```

Reply:
Thank you for the advice, I adjusted the assignment to be more immutable. The end result uses a ternary assignment because it needed to be a little bit more complex to prevent flipping between `true` and `false` during calls to the `createStateUpdate` method.

Reply:
Ah yeah. You are right.  Ternary works great. My brain was thinking of the if statement.. but you could also do it with or in the cause of just do true if one of them is true. `state.filterValueChanged || filter.length > 0` as well.